pkg/infra/main.go

```diff
@@ -16,25 +19,16 @@ Whereby the peer channels enable bidirectional p2p communication
 Enables peers to send and receive messages with broadcasting possible via non-blocking sends
 */
 
-const numLevels = 60
-
-const numAgents = 100
-
-const numAgentsRequired = numAgents * 2 / 5
-
-const monsterHealth = 100
-const monsterAttack = 240
-
 func main() {
-	agentMap, stateChannels, decisionChannels, globalState := initialise()
-	gameLoop(globalState, agentMap, decisionChannels, stateChannels)
+	agentMap, stateChannels, decisionChannels, globalState, gameConfig := initialise()
+	gameLoop(globalState, agentMap, decisionChannels, stateChannels, gameConfig)
 }
 
-func gameLoop(globalState state.State, agentMap map[uint]agent.Agent, decisionChannels map[uint]<-chan decision.Decision, stateChannels map[uint]chan<- state.State) {
-	for globalState.CurrentLevel = 0; globalState.CurrentLevel < numLevels; globalState.CurrentLevel++ {
+func gameLoop(globalState state.State, agentMap map[uint]agent.Agent, decisionChannels map[uint]<-chan decision.Decision, stateChannels map[uint]chan<- state.State, gameConfig config.GameConfig) {
```

Comment:
These functions are starting to take a lot of parameters... We should probably start sticking them into structs to tidy things up a bit.